src/field-builder.ts

```diff
@@ -145,3 +146,59 @@ fieldBuilderProto.effect = function effect({ effect = {}, resolve, ...options })
     }) as Resolver<any, any, any, any>,
   });
 };
+
+fieldBuilderProto.effectConnection = function connection(
+  { edgesNullable, nodeNullable, type, ...fieldOptions },
+  connectionOptionsOrRef = {} as never,
+  edgeOptionsOrRef = {} as never,
+) {
+  const connectionRef = connectionOptionsOrRef instanceof ObjectRef
+    ? connectionOptionsOrRef
+    : this.builder.objectRef<ConnectionShape<SchemaTypes, unknown, boolean>>(
+      'Unnamed connection',
+    );
```

Comment:
What does `Unnamed connection` meaning and doing if `connectionObjectOrRef` is not instance of `ObjectRef`? 

Here is my poor reasoning: `Unnamed connection` is just reference because `.implement(...)` is not called. If then, what happens in `this.builder.configStore.associateRefWithName(connectionRef, connectionName)`?

Reply:
basically, we need to ref before we know the name of the connection type.  Refs always need an associated name, but we don't have a good name for the type at this point.  If the field is never used, eg:

you do something like:

```
t.queryFields(t => {
  // this is not used
  t.connection({...})
  return {}
})
```
 
We never get the onFieldUse callback, and never update the ref this.builder.configStore.associateRefWithName, and you may see an error like `ref 'Unnamed connection' was not implemented`